src/datachain/lib/meta_formats.py

```diff
@@ -122,9 +118,9 @@ def read_meta(  # noqa: C901
             print(f"{model_output}")
         # Below 'spec' should be a dynamically converted DataModel from Pydantic
         if not spec:
-            local_vars: dict[str, Any] = {}
-            exec(model_output, globals(), local_vars)  # type: ignore[arg-type] # noqa: S102
-            spec = local_vars["spec"]
+            gl = globals()
+            exec(model_output, gl)  # noqa: S102
```

Comment:
We could avoid using `exec` and use `importlib` or `runpy.run_path` instead.
But `importlib` needs to update `sys.modules`, and `run_path` is not thread-safe, and makes a lot of modifications to `sys`.